content/blog/2023-03-17-build-grafana-dashboards-for-apache-skywalking-native-promql-support/index.md

```diff
@@ -0,0 +1,403 @@
+---
+title: "Build Grafana dashboards for Apache SkyWalking -- Native PromQL Support"
+date: 2023-03-16
+author: "Kai Wan"
+description: "."
+tags:
+- PromQL
+- Metric
+- Grafana
+---
+
+## Background
+As an application performance monitoring tool for distributed systems, Apache SkyWalking provides monitoring, tracing, diagnosing capabilities for distributed system in Cloud Native architecture.
+[Prometheus](https://prometheus.io/docs/introduction/overview/#what-is-prometheus) is an open-source systems monitoring and alerting toolkit with an active ecosystem.
+Especially Prometheus metrics receive widespread support through [exporters and integrations](https://prometheus.io/docs/instrumenting/exporters/#exporters-and-integrations).
+[PromQL](https://prometheus.io/docs/prometheus/latest/querying/basics/#querying-prometheus) as Prometheus Querying Language containing a set of expressions and expose HTTP API to read metrics.
+
+SkyWalking is support to ingest Prometheus metrics through [OpenTelemetry collector](https://opentelemetry.io/ecosystem/integrations/) and through the aggregate calculation of these metrics
+to provide a variety of systems monitoring, such as Linux Monitoring and Kubernetes monitoring.
+SkyWalking already provides [native UI](https://skywalking.apache.org/docs/main/next/en/ui/readme/) and [GraphQL API](https://skywalking.apache.org/docs/main/next/en/api/query-protocol/) for users.
+But as designed to provide wider ecological integration capabilities, since 9.4.0, it provides PromQL Service, the third-party systems or visualization platforms that already support PromQL (such as Grafana),
+could obtain metrics through it. SkyWalking users will benefit from it when they integrate with different systems.
+
+## What is PromQL Service in SkyWalking?
+PromQL Service is a query engine on the top of SkyWalking native GraphQL query, with additional query stage calculation capabilities powered by Prometheus expressions.
+It can accept PromQL HTTP API requests, parse Prometheus expression, and transform between Prometheus metrics and SkyWalking metrics.
```

Comment:
```suggestion
It can accept PromQL HTTP API requests, parse Prometheus expressions, and transform between Prometheus metrics and SkyWalking metrics.
```